### PR TITLE
[BUGFIX beta] cleanup `setPath`

### PR DESCRIPTION
**Pull request by bekzod**
cleaned up `setPath` using simpler logic, which is probably faster as test running a bit faster too

### CODE REVIEW

File: packages/ember-metal/lib/property_set.js [hunk 1]
```diff
@@ -100,16 +100,13 @@ if (MANDATORY_SETTER) {
...
+  let parts = path.split('.');
+  let keyName = parts.pop();
 
   // get the first part of the part
-  path = (path === keyName) ? keyName : path.slice(0, path.length - (keyName.length + 1));
+  path = (path === keyName) ? keyName : parts.join('.');
 
-  // unless the path is this, look up the first part to
-  // get the root
-  if (path !== 'this') {
```
Comment (bekzod) [inline packages/ember-metal/lib/property_set.js]: there is already assertion to not allow that https://github.com/bekzod/ember.js/blob/57f47143694d0db1197bfec0a582a40defbb0ca9/packages/ember-metal/lib/property_set.js#L43

File: packages/ember-metal/lib/property_set.js [hunk 2]
```diff
@@ -99,22 +99,17 @@ if (MANDATORY_SETTER) {
...
 function setPath(root, path, value, tolerant) {
-  // get the last part of the path
-  let keyName = path.slice(path.lastIndexOf('.') + 1);
-
-  // get the first part of the part
-  path = (path === keyName) ? keyName : path.slice(0, path.length - (keyName.length + 1));
-
-  // unless the path is this, look up the first part to
-  // get the root
-  if (path !== 'this') {
```
Comment (bekzod) [inline packages/ember-metal/lib/property_set.js]: there is already assertion to not allow that https://github.com/bekzod/ember.js/blob/57f47143694d0db1197bfec0a582a40defbb0ca9/packages/ember-metal/lib/property_set.js#L43

Comment (stefanpenner) [inline packages/ember-metal/lib/property_set.js]: @krisselden r?

File: packages/ember-metal/lib/property_set.js [hunk 3]
```diff
@@ -99,22 +99,17 @@ if (MANDATORY_SETTER) {
...
-    root = getPath(root, path);
-  }
+  let parts = path.split('.');
+  let keyName = parts.pop();
 
   if (!keyName || keyName.length === 0) {
     throw new EmberError('Property set failed: You passed an empty path');
```
Comment (krisselden) [inline packages/ember-metal/lib/property_set.js]: we should probably just assert this.

File: packages/ember-metal/lib/property_set.js [hunk 4]
```diff
@@ -99,22 +99,17 @@ if (MANDATORY_SETTER) {
...
   }
 
+  path = parts.length > 0 ? parts.join('.') : keyName;
+
+  root = getPath(root, path);
```
Comment (krisselden) [inline packages/ember-metal/lib/property_set.js]: we shouldn't mutate arguments, we should use a new var.

File: packages/ember-metal/lib/property_set.js (hunk 2 above)
Comment (krisselden) [inline packages/ember-metal/lib/property_set.js]: This is a very good catch, we removed support for this.paths, it was a workaround for capitalized paths back when they meant global (which never fully worked anyway).

**Review by krisselden — APPROVED** [review]
thank you for this, I kind of what this to be marked for beta, @stefanpenner thoughts?

**Comment by stefanpenner** [comment]
@bekzod we can likely sneak this into beta, if you could update the commit message to be prefixed with `[BUGFIX beta]` that would be great.

**Pull request merged**